packages/auth/src/grant/routes.ts

```diff
@@ -137,7 +138,18 @@ async function createPendingGrant(
   const { body } = ctx.request
   const { grantService, interactionService, config } = deps
   if (!body.interact) {
-    ctx.throw(400, { error: 'interaction_required' })
+    ctx.throw(400, 'interaction_required', { error: 'interaction_required' })
+  }
+
+  if (
+    body.access_token.access.every(
+      (item) =>
+        (item.actions.includes(AccessAction.ReadAll) ||
+          item.actions.includes(AccessAction.ReadAll)) &&
+        item.identifier !== ''
+    )
+  ) {
```

Comment:
Can we simplify this to:
```suggestion
  if (body.access_token.access.find((item) => !item.identifier)) {
```

If this:
> check whether when a grant that requires interaction is requested, the identifier is present

is what we want?

Reply:
I don't think that is what we want. This condition would also fire on accesses that don't need an identifier. But I want to make sure that the identifier is within the access that triggered the interaction. (This is so hard to explain...)